j5/components/gpio_pin.py

```diff
@@ -136,8 +133,7 @@ def mode(self, pin_mode: GPIOPinMode) -> None:
         """Set the hardware mode of this pin."""
         if pin_mode not in self._supported_modes:
             raise NotSupportedByHardwareError(
-                f"Pin {self._identifier} on {str(self._board)} \
-                does not support {str(pin_mode)}.",
+                f"Pin {self._identifier} does not support {str(pin_mode)}.",
```

Comment:
I think that losing the board that this pin is on is a slight loss, but that this also opens up a `Component` to be independent of a `Board`, which I approve of.